lib/console.js

```diff
@@ -26,13 +26,18 @@ const util = require('util');
 const kCounts = Symbol('counts');
 
 // Track amount of indentation required via `console.group()`.
-const kGroupIndent = Symbol('groupIndent');
+const kGroupIndent = Symbol('kGroupIndent');
+
+const kFormatForStderr = Symbol('kFormatForStderr');
+const kFormatForStdout = Symbol('kFormatForStdout');
+const kGetInspectOptions = Symbol('kGetInspectOptions');
+const kColorMode = Symbol('kColorMode');
 
 let MAX_STACK_MESSAGE;
 
-function Console(stdout, stderr, ignoreErrors = true) {
+function Console(stdout, stderr, ignoreErrors = true, colorMode = 'auto') {
```

Comment:
This is the fourth positional argument to the constructor, maybe we should consider an options object at some point (unless it is foreseeable that we won't add more options to the constructor).

Reply:
Yea, I’m happy to do that if you think it’s a good idea.

I don’t think an ergonomic API matters all that much here, because almost nobody is constructing their own `console` objects, but I could totally see the argument being made for it.

Reply:
I shouldn't argue for either side, I never use the constructor, it's just a general concern I'd have with any API. If you type-check `colorMode` strictly, we can still replace it with an options object later.

Reply:
I think if we do go for an options argument, it should probably encompass all arguments here – so, `new Console({ stdout, stderr, ignoreErrors, colorMode, ... })`.

I’m not sure how well that would work, though, because all of these properties could just be accidentally set on the `stdout` object itself…

Reply:
An options object would be best, I think.

Reply:
@jasnell Do you have thoughts on *how* we would detect an options object?

I guess we could try to see whether it’s a duck and has a `write()` property…

Reply:
Maybe `Console(stdout, [stderr, [options]])`? That kind of makes sense as `ignoreErrors` and `colorMode` have reasonable defaults and can be distinguished from an object.

Reply:
If going with the options property I would use the single argument strategy you suggest above and check if the first argument is an object with at least the stdin property set that does not have a write or maybe _write property whose value is a function.